devito/finite_differences/differentiable.py

```diff
@@ -645,6 +645,47 @@ def __str__(self):
     __repr__ = __str__
 
 
+class ComplexPart(Differentiable, sympy.core.function.Application):
+    """Abstract class for `Real`, `Imag`, or `Conj` of an expression"""
+    _name = None
+
+    def __new__(cls, *args, **kwargs):
+        if len(args) != 1:
+            raise ValueError(f"{cls.__name__} is constructed with exactly one arg;"
```

Comment:
I'd rewrite it as 

* "is constructed with" -> "expects";
* "were supplied" -> "were supplied instead"